packages/gestalt/src/TypeaheadInputField.js

```diff
@@ -0,0 +1,206 @@
+  placeholder?: string,
+  size?: 'md' | 'lg',
+  value?: string,
+  label?: string,
+  forwardedRef?: React.Ref<'input'>,
+|};
+
+const InputField = ({
+  id,
+  label = id,
+  onBlur,
+  onChange,
+  onClear,
+  onKeyNavigation,
+  onFocus,
+  setContainer,
+  placeholder,
+  size = 'md',
+  value,
+  forwardedRef,
+}: Props): Node => {
+  const [hovered, setHovered] = React.useState<boolean>(false);
+
+  const handleChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    onChange({
+      value: event.currentTarget.value,
+      event,
+    });
+  };
+
+  const handleClear = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    onChange({ value: '', event });
+
+    if (onClear) onClear();
+  };
+  const handleMouseEnter = () => setHovered(true);
+
+  const handleMouseLeave = () => setHovered(false);
+
+  const handleFocus = (event: SyntheticFocusEvent<HTMLInputElement>) => {
+    if (onFocus) {
+      onFocus({
+        value: event.currentTarget.value,
+        event,
+      });
+    }
+  };
+
+  const handleBlur = (event: SyntheticFocusEvent<HTMLInputElement>) => {
+    if (onBlur) {
+      onBlur({ event });
+    }
+  };
+
+  const handleClick = (event: SyntheticFocusEvent<HTMLInputElement>) => {
+    handleFocus(event);
+    setContainer(true);
+  };
+
+  const handleKeyNavigation = (
+    event: SyntheticKeyboardEvent<HTMLInputElement>
+  ) => {
+    setContainer(true);
+    const KEYS = {
+      UP: -1,
+      DOWN: 1,
+      ENTER: 0,
+    };
+    // Up Arrow
+    if (event.keyCode === 38) {
+      onKeyNavigation(event, KEYS.UP);
+    }
+    // Down Arrow
+    else if (event.keyCode === 40) {
+      onKeyNavigation(event, KEYS.DOWN);
+    }
+    // Enter Key
+    else if (event.keyCode === 13) {
+      onKeyNavigation(event, KEYS.ENTER);
+    }
+  };
+
+  const hasValue = value && value?.length > 0;
+
+  const className = classnames(styles.input, {
+    [layout.medium]: size === 'md',
+    [layout.large]: size === 'lg',
+    [styles.inputActive]: true,
+    [styles.inputHovered]: hovered,
+    [typeaheadStyle.inputRadius]: true,
+  });
+
+  const clearButtonSize = size === 'lg' ? 24 : 20;
+  const clearIconSize = size === 'lg' ? 12 : 10;
+
+  return (
+    <>
+      {label && <FormLabel id={id} label={label} />}
+      <Box
+        alignItems="center"
+        display="flex"
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}
+        position="relative"
+      >
+        <input
+          ref={forwardedRef}
+          autoComplete="off"
+          aria-label={label}
+          className={className}
+          id={id}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onClick={() => setContainer(true)}
+          onChange={handleChange}
+          placeholder={placeholder}
+          type="text"
+          value={value}
+          onKeyDown={handleKeyNavigation}
+        />
+
+        <button
```

Comment:
@simkessy please update the `x` button so it also works with RTL. See #970 how we did it for `SearchField`

SearchField (correct)
![image](https://user-images.githubusercontent.com/127199/87580739-ce795280-c68c-11ea-8341-0a7c56877132.png)

Typeahead
![image](https://user-images.githubusercontent.com/127199/87580802-e355e600-c68c-11ea-89c6-1feb67130c8d.png)